tests/unit/blocklist_test.py

```diff
@@ -60,6 +60,28 @@ def test_file_loading(self, tmp_path, line, reason):
 
         assert blocklist.domains == ({"example.com": reason} if reason else {})
 
+    @pytest.mark.parametrize(
+        "line,reason",
+        (
+            ("*.example.com publisher-blocked", Blocklist.Reason.PUBLISHER_BLOCKED),
+            ("*.example.com rubbish", Blocklist.Reason.OTHER),
+            ("*.example.com", None),
+        ),
+    )
+    def test_file_loading_wildcards(self, tmp_path, line, reason):
+        filename = tmp_path / "blocklist.txt"
+        filename.write_text(line)
+
+        blocklist = Blocklist(filename)
+
+        if not reason:
+            assert blocklist.patterns == {}
+        else:
+            assert len(blocklist.patterns) == 1
+            ((domain, found_reason),) = blocklist.patterns.items()
+            assert found_reason == reason
+            assert domain.pattern == r"^.*\.example\.com$"
```

Comment:
Could have used an Any matcher here, but it's a bit of a pain to get the right type.